src/material/table/table-data-source.ts

```diff
@@ -103,18 +103,16 @@ export class MatTableDataSource<T, P extends MatPaginator = MatPaginator> extend
    * Instance of the MatSort directive used by the table to control its sorting. Sort changes
    * emitted by the MatSort will trigger an update to the table's rendered data.
    */
-  get sort(): MatSort | null {
+  get sort(): MatSort | null | undefined {
```

Comment:
This is technically a public API change. It might be better to change the setter to `|| null` so the runtime matches the type.